### PR TITLE
General Refactor and Documentation Updates

### PR DESCRIPTION
**Pull request by ransom4real**
### Context

We do not catch missing env vars earlier in the app loading hence we get failures during initialisation without useful error messages with help on how to fix. Also the documentation for testing and running the app is missing a few bits and bobs

### Changes proposed in this pull request

- Update app initialisation to catch missing env vars earlier than later with a clearer failure message and reference to appropriate README sections
- Add precompile script to test section of readme
- Updated build css script to quiet warnings
- Added README section for handling basic auth where feature flag for service open is set to false

### Guidance to review

Verify documentation is accurate and useful
Review error message for missing env var to ensure it is clear

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: adr/00008-handle-missing-env-vars.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# 8. handle-missing-env-vars
```
Comment (tvararu) [inline adr/00008-handle-missing-env-vars.md:1]: Discussed offline, but this change shouldn't require an ADR as it's quite small :+1:

File: README.md [hunk 2]
```diff
@@ -123,6 +123,10 @@ Run the application on `http://localhost:3000`:
 bin/dev
 ```
 
+### Basic Auth
+
+Due to the `Service open` feature flag which is set to `false` by default across all environments except Production, you will be asked to sign in with basic auth. You can find credentials in the .env._environment_ file stored in the ENV variables `SUPPORT_USERNAME` and `SUPPORT_PASSWORD`. You can toggle the `Service open` feature flag to `true` when developing locally to spare you having to always use basic auth.
```
Comment (tvararu) [inline README.md:128]:
```suggestion
Due to the `Service open` feature flag which is set to `false` by default across all environments except Production, you will be asked to sign in with basic auth. You can find credentials in the .env._environment_ file stored in the ENV variables `SUPPORT_USERNAME` and `SUPPORT_PASSWORD`. You can toggle the `Service open` feature flag to `true` when developing locally to switch off basic auth.
```

File: README.md [hunk 3]
```diff
@@ -140,6 +144,8 @@ GOVUK_NOTIFY_API_KEY=theo__local_test-abcefgh-1234-abcdefgh
 When you send an email locally, the email should appear in the message log in
 the Notify dashboard in the `API integration` section.
 
+_Note:_ It is acceptable to set `GOVUK_NOTIFY_API_KEY=fake-key` when trying to quickly run the project locally
```
Comment (tvararu) [inline README.md:147]:
```suggestion
_Note:_ You can set `GOVUK_NOTIFY_API_KEY=fake-key` when running locally if you don't need to use Notify.
```

File: README.md [hunk 4]
```diff
@@ -163,6 +169,14 @@ bin/lint
 To run the tests (requires Chrome due to
 [cuprite](https://github.com/rubycdp/cuprite)):
 
+Run the command below to allow assets to be compiled and cached up front as tests that utilise cuprite will need this
+
+```bash
+bin/rails assets:precompile
+```
+
+And then
```
Comment (tvararu) [inline README.md:178]:
```suggestion
And then:
```

File: README.md [hunk 5]
```diff
@@ -163,6 +169,14 @@ bin/lint
 To run the tests (requires Chrome due to
 [cuprite](https://github.com/rubycdp/cuprite)):
 
+Run the command below to allow assets to be compiled and cached up front as tests that utilise cuprite will need this
```
Comment (tvararu) [inline README.md:172]:
```suggestion
To compile assets up front (needed by the end to end tests):
```

File: config/application.rb [hunk 6]
```diff
@@ -42,5 +42,11 @@ class Application < Rails::Application
     config.active_job.queue_adapter = :sidekiq
 
     config.exceptions_app = routes
+
+    # Fetch Notify API Key if configured or raise exception with descriptive message
```
Comment (tvararu) [inline config/application.rb:46]: I think the code is descriptive on its own without the comment:

```suggestion
```

**Review by tvararu — COMMENTED** [review]
Good work! Just a few suggestions for content changes, making the voice in the README more consistent.

Could you also redo the commits to follow the GDS styleguide? https://gds-way.cloudapps.digital/standards/source-code/working-with-git.html#commits

Just an example of how we could phrase this work:

```
Update documentation for first run setup

- Add precompile script to test section of readme
- Add ADR for handling missing environment variables
- Update build css script to quiet warnings
- Add README section for handling basic auth where feature flag for service open is set to false
```

You can also squash away the `prettier` linting fix commit. :+1:

File: adr/00008-handle-missing-env-vars.md (hunk 1 above)
Comment (ransom4real) [reply adr/00008-handle-missing-env-vars.md:1]: Removed

File: config/application.rb (hunk 6 above)
Comment (ransom4real) [reply config/application.rb:46]: Comment removed

File: README.md (hunk 4 above)
Comment (ransom4real) [reply README.md:178]: Suggestion updated

File: README.md (hunk 3 above)
Comment (ransom4real) [reply README.md:147]: Suggestion updated

File: README.md (hunk 2 above)
Comment (ransom4real) [reply README.md:128]: Suggestion updated

**Review by tvararu — APPROVED** [review]

**Pull request merged**